.github/workflows/testing-and-deployment.yml

```diff
@@ -191,7 +191,7 @@ jobs:
           pip list
 
       - name: Unit Testing
-        run: xvfb-run coverage run -m --source=pyvista --module pytest --skip_image_cache_vtk8 --fail_extra_image_cache -v
+        run: xvfb-run coverage run -m --source=pyvista --module pytest --skip_image_cache_vtk8 --fail_extra_image_cache -v --add_missing_images
```

Comment:
Could you give an explanation of the reason for adding this for me?

Reply:
I'm getting an error when adding a new test that needs image comparison:
>FAILED tests/plotting/test_plotting.py::test_axes_actor_properties - RuntimeError: tests/plotting/image_cache/axes_actor_properties.png does not exist in image cache

This flag should generate the missing images without the need of resetting the whole cache:
> `--add_missing_images` adds any missing images from the test run to the cache.

but is not working in the repository for some reason, I'm trying to find out why.

Reply:
I think you need to [add an image for regression tests](https://docs.pyvista.org/extras/developer_notes.html#notes-regarding-image-regression-testing).

Reply:
The image was not being generated because the flag `--fail_extra_image` prevents the code from reaching the  `--add_missing_images` if branch. You can see it in this piece of code from `pytest-pyvista`:

```python

image_filename = os.path.join(self.cache_dir, test_name[5:] + ".png")
if not os.path.isfile(image_filename) and self.fail_extra_image_cache:
    raise RuntimeError(f"{image_filename} does not exist in image cache")

if self.add_missing_images and not os.path.isfile(image_filename):
    plotter.screenshot(image_filename)

```

This is probably an undesired behaviour in `pytest-pyvista`, I will open an issue to discuss it there.

Reply:
`--fail_extra_image_cache` `--add_missing_images` are incompatible together. You can either fail on missing or generate new images. Further, we always want the CI to fail on missing and encourage developers to check in new images with the code

Reply:
I think this should be properly documented in any case. Also, maybe `pytest-pyvista` could raise at least a warning regarding you have both flags active.